core/app/models/spree/return_item.rb

```diff
@@ -233,10 +235,12 @@ def currency
 
     def process_inventory_unit!
       inventory_unit.return!
-
       if customer_return
         customer_return.stock_location.restock(inventory_unit.variant, 1, customer_return) if should_restock?
-        customer_return.process_return!
+        unless skip_customer_return_processing
```

Comment:
I think we should also test that a deprecation warning is raised in some circumstances.

Reply:
Makes sense, I added this spec: https://github.com/solidusio/solidus/pull/3199/files#diff-0f55d3185362ecf4697cad5429703bd0